lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -789,7 +789,12 @@ SILCombiner::buildConcreteOpenedExistentialInfoFromSoleConformingType(
   ConcreteExistentialInfo &SoleCEI = *COAI.CEI;
   assert(SoleCEI.isValid());
 
-  if (SoleCEI.ConcreteValue)
+  // In ossa, don not use the concrete value previously found as it maybe
+  // forward consumed and not available at the apply using the opened
+  // existential.
+  if (SoleCEI.ConcreteValue &&
+      (!F->hasOwnership() ||
+       SoleCEI.ConcreteValue->getOwnershipKind() != OwnershipKind::Owned))
```

Comment:
Hm, can we still optimize if the `open_existential_ref` is the only use of the `init_existential_ref`? 
Then this pair of instructions can be removed and no copy is required.

Reply:
`open_existential_ref` also needs to have a single user i.e apply which will be converted to concrete value for this to work. The optimization also supports introducing unconditional cast for `open_existential_ref` to the concrete type and using the cast value in the apply. I'll try that since we at least get a concrete apply in that case.